assets/js/phoenix_live_view.js

```diff
@@ -1271,6 +1271,21 @@ export let DOM = {
     } else {
       return true
     }
+  },
+
+  cleanChildNodes(container, phxUpdate){
+    if (DOM.isPhxUpdate(container, phxUpdate, ["append", "prepend"])) {
+      Array.from(container.childNodes).forEach(childNode => {
+        if (childNode.nodeType == Node.TEXT_NODE && childNode.nodeValue.trim() == "") {
```

Comment:
I think we can remove them even if they have text. We just need to warn if they have text too. So all non text nodes and all text nodes with contents are warned. WDYT?

Reply:
This code did that but it wasn't very clear in the intent. I rewrote it to be clearer.